### PR TITLE
Feat: `Regionalization` Doc Storybook

### PR DESCRIPTION
**Pull request by renatamottam**
## What's the purpose of this pull request?
Creates a page for `Regionalization` feature to list all its components and how to use it.
Also, this PR brings changes from #104.

## How does it work?
Each feature will now have its own page and should:
1. Explain its usage and purpose
2. List components with descriptions
3. Add possibility to test it

|Preview|
|-|
||

## How to test it?
See [preview link](https://nextjs-store-storybook-git-feat-theming-region-f57660-faststore.vercel.app/?path=/docs/features-regionalization-overview--default-button)

## Checklist
**Changelog**
- [x] Added an entry in the `CHANGELOG.md` at the beginning of its due section. [The latest version should comes first](https://keepachangelog.com/en/1.0.0/#:~:text=The%20latest%20version%20comes%20first.).
- [x] Added the PR number with the PR link at the entry in the `CHANGELOG.md`. E.g., *New items in the `pull_request_template.md` ([#4](https://github.com/vtex-sites/nextjs.store/pull/4))* 

**PR Description**
- [x] Added a label according to the PR goal - `Breaking change`, `Enhancement`, `Bug` or `Chore`.
- [x] Added the component, hook, or pathname in-between backticks (``) *- If applicable*. E.g., *`ComponentName` component*.

**Documentation**
- [x] PR description
- [x] Added to/Updated the Storybook - *if applicable*.
- [x] For documentation changes, ping @ carolinamenezes, @ PedroAntunesCosta or @ Mariana-Caetano to review and update.

### CODE REVIEW

File: src/components/regionalization/Regionalization.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+      </SessionProvider>
+    </UIProvider>
+  )
+}
+
+<header>
+
+# Regionalization
+
+This feature enables customers to Set a location using a valid Postal Code.
```
Comment (hellofanny) [inline src/components/regionalization/Regionalization.stories.mdx:54]:
```suggestion
This feature enables customers to set a location using a valid Postal Code.
```

File: src/components/regionalization/Regionalization.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+</header>
+
+## Overview
+
+1. Intentionally, a user Sets its location using a Postal Code
+2. Updates Product Catalog
+3. Gets notified if there's product conflict at minicart
```
Comment (hellofanny) [inline src/components/regionalization/Regionalization.stories.mdx:62]: I think we need an introduction before this list 🙈

File: src/components/regionalization/Regionalization.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+---
+
+## Usage
+
+Import RegionalizationComponents `from '../components/regionalization/...'`
```
Comment (hellofanny) [inline src/components/regionalization/Regionalization.stories.mdx:68]:
```suggestion
Import Regionalization<Component> `from '../components/regionalization/<Component>'`
```

File: src/components/regionalization/RegionalizationBar/regionalization-bar.module.scss [hunk 4]
```diff
@@ -0,0 +1,58 @@
+@import "src/styles/scaffold";
+
+.fs-regionalization-bar {
+  // --------------------------------------------------------
+  // Design Tokens for Incentives
```
Comment (hellofanny) [inline src/components/regionalization/RegionalizationBar/regionalization-bar.module.scss:5]:
```suggestion
  // Design Tokens for Regionalization Bar
```

File: src/components/regionalization/Regionalization.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+          <UIProvider>
+            <RegionalizationBar />
+            <TemplateRegionalizationModal />
+          </UIProvider>
+        </SessionProvider>
+      </div>
+      <article className="sbdocs-list-text">
+        <h3>RegionalizationBar</h3>
+        <p>
+          The Bar is a similar proposal for the button, usually used on mobile
```
Comment (hellofanny) [inline src/components/regionalization/Regionalization.stories.mdx:100]:
```suggestion
          The Bar has a similar proposal for the button, usually used on mobile
```

File: src/components/regionalization/RegionalizationBar/RegionalizationBar.stories.mdx [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+    isColor
+  />
+  <TokenRow token="--fs-regionalization-bar-border" value="0" />
+</TokenTable>
+
+---
+
+## Nested Elements
+
+### CTA
```
Comment (hellofanny) [inline src/components/regionalization/RegionalizationBar/RegionalizationBar.stories.mdx:85]:
```suggestion
### CTA (Call To Action)
```

Just to help.

File: src/components/regionalization/RegionalizationBar/regionalization-bar.module.scss [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+  // --------------------------------------------------------
+  // Design Tokens for Regionalization Bar
+  // --------------------------------------------------------
+
+  // Default properties
+  --fs-regionalization-bar-width                    : 100%;
+  --fs-regionalization-bar-padding                  : var(--fs-spacing-0) var(--fs-spacing-3);
+  --fs-regionalization-bar-text-color               : var(--fs-color-text-display);
+  --fs-regionalization-bar-background-color         : var(--fs-color-body-bkg);
+  --fs-regionalization-bar-border                   : 0;
```
Comment (hellofanny) [inline src/components/regionalization/RegionalizationBar/regionalization-bar.module.scss:13]: `0` or `none` hahaha. Do we have a pattern for it?

File: src/components/regionalization/RegionalizationBar/regionalization-bar.module.scss [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+  }
+
+  [data-fs-regionalization-bar-cta] {
+    margin-left: var(--fs-regionalization-bar-cta-margin-left);
+    text-decoration: var(--fs-regionalization-bar-cta-text-decoration);
+  }
+
+  [data-fs-regionalization-bar-postal-code] {
+    margin-right: var(--fs-regionalization-bar-postal-code-margin-right);
+    font-weight: var(--fs-text-weight-regular);
```
Comment (hellofanny) [inline src/components/regionalization/RegionalizationBar/regionalization-bar.module.scss:48]:
```suggestion
    font-weight: var(--fs-regionalization-bar-postal-code-font-weight);
```

File: src/components/regionalization/RegionalizationBar/regionalization-bar.module.scss [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+  }
+
+  [data-fs-regionalization-bar-message] {
+    margin-right: var(--fs-regionalization-bar-message-margin-right);
+    font-weight: var(--fs-text-weight-regular);
```
Comment (hellofanny) [inline src/components/regionalization/RegionalizationBar/regionalization-bar.module.scss:53]:
```suggestion
    font-weight: var(--fs-regionalization-bar-message-font-weight);
```

Comment (hellofanny) [reply src/components/regionalization/RegionalizationBar/regionalization-bar.module.scss:53]: btw, do we really need those `font-weight` tokens here? 🤔

**Review by hellofanny — APPROVED** [review]
Pre-approved ✅ (just waiting the link to the `RegionalizationModal` when available).
This doc looks great!! Well done! 🤩 
Small suggestion: rename `Default` to `Overview`. I think it fits better; what do you think?

image

File: src/components/regionalization/Regionalization.stories.mdx [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+          An actionable{' '}
+          <a href="../?path=/docs/atoms-inputtext--actionable">InputText</a>{' '}
+          able to check the Postal Code authenticity.
+        </p>
+      </article>
+    </li>
+    <li>
+      <div>
+        <img
+          src="https://user-images.githubusercontent.com/11613011/172647856-b7b733ec-870f-45df-b26f-476456179a7e.png"
```
Comment (hellofanny) [inline src/components/regionalization/Regionalization.stories.mdx:133]:
```suggestion
          src="https://vtexhelp.vtexassets.com/assets/docs/src/regionalization-modal-example___3516d5267008e3311ea30ae12b21eab4.png"
```

**Comment by hellofanny** [comment]
Looks [beautifuuuul](https://nextjs-store-storybook-git-feat-theming-region-f57660-faststore.vercel.app/?path=/docs/features-regionalization-overview--default-button)! 🤩

**Review by ArthurTriis1 — APPROVED** [review]
Awesome!

**Pull request merged**